### PR TITLE
RHDEVDOCS-4310 - clarify confusing statements about custom Prometheus support

### LINKED ISSUE #40244: [enterprise-4.7] Issue in file monitoring/exposing-custom-application-metrics-for-autoscaling.adoc
### Which section(s) is the issue in? 
Exposing custom application metrics for horizontal pod autoscaling

### What needs fixing?
[Here ](https://docs.openshift.com/container-platform/4.7/monitoring/exposing-custom-application-metrics-for-autoscaling.html#exposing-custom-application-metrics-for-horizontal-pod-autoscaling_exposing-custom-application-metrics-for-autoscaling)it says 
_**Prerequisites**
You have a custom Prometheus instance installed. In this example, it is presumed that Prometheus was installed in a user-defined custom-prometheus project._

But [here ](https://docs.openshift.com/container-platform/4.7/monitoring/configuring-the-monitoring-stack.html#support-considerations_configuring-the-monitoring-stack)
It says 
_The following modifications are explicitly not supported:_
**Installing custom Prometheus instances on OpenShift Container Platform.**

### PR DESCRIPTION
**Pull request by bburt-rh**
Summary:

- Aligned team: DevTools
- For branches: 4.6, 4.7 ONLY
- Jira: https://issues.redhat.com/browse/RHDEVDOCS-4310
- Direct link to doc preview (RH VPN access required):
- - http://file.rdu.redhat.com/bburt/RHDEVDOCS-4310-clarify-contradictory-guidance-in-autoscaling-topic/monitoring/configuring-the-monitoring-stack.html#support-considerations_configuring-the-monitoring-stack
- - http://file.rdu.redhat.com/bburt/RHDEVDOCS-4310-clarify-contradictory-guidance-in-autoscaling-topic/monitoring/exposing-custom-application-metrics-for-autoscaling.html#exposing-custom-application-metrics-for-horizontal-pod-autoscaling_exposing-custom-application-metrics-for-autoscaling
- SME review: @simonpasquier 
- QE review: @juzhao  
- Peer review: @gabriel-rh

### CODE REVIEW

File: modules/monitoring-exposing-custom-application-metrics-for-horizontal-pod-autoscaling.adoc [hunk 1]
```diff
@@ -11,11 +11,12 @@ You can use the `prometheus-adapter` resource to expose custom application metri
...
 
-* You have a custom Prometheus instance installed. In this example, it is presumed that Prometheus was installed in a user-defined `custom-prometheus` project.
+* You have a custom Prometheus instance installed as a Prometheus pod managed by a deployment or statefulset but not by a Prometheus custom resource (CR).  
+* You have installed the custom Prometheus instance in a user-defined `custom-prometheus` project.
 +
-[NOTE]
+[IMPORTANT]
 ====
-Custom Prometheus instances and the Prometheus Operator installed through Operator Lifecycle Manager (OLM) can cause issues with user-defined workload monitoring if it is enabled. Custom Prometheus instances are not supported in {product-title}.
+Custom Prometheus instances and the Prometheus Operator installed through Operator Lifecycle Manager (OLM) are not compatible with user-defined monitoring if user-defined monitoring is enabled. Therefore, custom Prometheus instances that are installed as a Prometheus custom resource (CR) managed by the Prometheus Operator are not supported in {product-title}.
```
Comment (simonpasquier) [inline modules/monitoring-exposing-custom-application-metrics-for-horizontal-pod-autoscaling.adoc:19]:
```suggestion
Custom Prometheus instances and the Prometheus Operator installed through Operator Lifecycle Manager (OLM) are not compatible with user-defined monitoring if user-defined monitoring is enabled. Therefore, custom Prometheus instances that are installed as a Prometheus custom resource (CR) managed by the OLM Prometheus Operator are not supported in {product-title}.
```

Comment (bburt-rh) [reply modules/monitoring-exposing-custom-application-metrics-for-horizontal-pod-autoscaling.adoc]: Revised per your suggestion

File: modules/monitoring-exposing-custom-application-metrics-for-horizontal-pod-autoscaling.adoc [hunk 2]
```diff
@@ -11,11 +11,12 @@ You can use the `prometheus-adapter` resource to expose custom application metri
 
 .Prerequisites
 
-* You have a custom Prometheus instance installed. In this example, it is presumed that Prometheus was installed in a user-defined `custom-prometheus` project.
+* You have a custom Prometheus instance installed as a Prometheus pod managed by a deployment or statefulset but not by a Prometheus custom resource (CR).
```
Comment (gabriel-rh) [inline modules/monitoring-exposing-custom-application-metrics-for-horizontal-pod-autoscaling.adoc:14]: Should it be stateful set as two separate words?

File: modules/monitoring-exposing-custom-application-metrics-for-horizontal-pod-autoscaling.adoc [hunk 3]
```diff
@@ -11,11 +11,12 @@ You can use the `prometheus-adapter` resource to expose custom application metri
...
 
-* You have a custom Prometheus instance installed. In this example, it is presumed that Prometheus was installed in a user-defined `custom-prometheus` project.
+* You have a custom Prometheus instance installed as a Prometheus pod managed by a deployment or statefulset but not by a Prometheus custom resource (CR).  
+* You have installed the custom Prometheus instance in a user-defined `custom-prometheus` project.
 +
-[NOTE]
+[IMPORTANT]
 ====
-Custom Prometheus instances and the Prometheus Operator installed through Operator Lifecycle Manager (OLM) can cause issues with user-defined workload monitoring if it is enabled. Custom Prometheus instances are not supported in {product-title}.
+Custom Prometheus instances and the Prometheus Operator installed through Operator Lifecycle Manager (OLM) are not compatible with user-defined monitoring if user-defined monitoring is enabled. Therefore, custom Prometheus instances that are installed as a Prometheus custom resource (CR) managed by the OLM Prometheus Operator are not supported in {product-title}.
```
Comment (gabriel-rh) [inline modules/monitoring-exposing-custom-application-metrics-for-horizontal-pod-autoscaling.adoc:19]:
```suggestion
Custom Prometheus instances and the Prometheus Operator installed through Operator Lifecycle Manager (OLM) are not compatible with user-defined monitoring, if user-defined monitoring is enabled. Therefore, custom Prometheus instances that are installed as a Prometheus custom resource (CR) managed by the OLM Prometheus Operator are not supported in {product-title}.
```

Comment (gabriel-rh) [reply modules/monitoring-exposing-custom-application-metrics-for-horizontal-pod-autoscaling.adoc:19]: Maybe add a comma here, to make parsing the sentence easier?

**Review by gabriel-rh — COMMENTED** [review]
Looks good - some very minor suggestions

File: modules/monitoring-exposing-custom-application-metrics-for-horizontal-pod-autoscaling.adoc (hunk 2 above)
Comment (bburt-rh) [reply modules/monitoring-exposing-custom-application-metrics-for-horizontal-pod-autoscaling.adoc:14]: No, but it should be `StatefulSet`. Good catch! Fixed.

https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/

File: modules/monitoring-exposing-custom-application-metrics-for-horizontal-pod-autoscaling.adoc (hunk 3 above)
Comment (bburt-rh) [reply modules/monitoring-exposing-custom-application-metrics-for-horizontal-pod-autoscaling.adoc:19]: I've revised it to put the "if" clause at the beginning to make parsing the meaning easier.

File: modules/monitoring-support-considerations.adoc [hunk 4]
```diff
@@ -18,7 +18,7 @@ The Alertmanager configuration is deployed as a secret resource in the `openshif
 * *Modifying resources of the stack.* The {product-title} monitoring stack ensures its resources are always in the state it expects them to be. If they are modified, the stack will reset them.
 * *Deploying user-defined workloads to `openshift-&#42;`, and `kube-&#42;` projects.* These projects are reserved for Red Hat provided components and they should not be used for user-defined workloads.
 * *Modifying the monitoring stack Grafana instance.*
-* *Installing custom Prometheus instances on {product-title}.*
+* *Installing custom Prometheus instances on {product-title}.* A custom instance is a Prometheus custom resource (CR) managed by the Prometheus Operator.
```
Comment (juzhao) [inline modules/monitoring-support-considerations.adoc:21]: managed by the Prometheus Operator
->
managed by the OLM Prometheus Operator

Comment (simonpasquier) [reply modules/monitoring-support-considerations.adoc:21]: "managed by the Prometheus Operator" is fine by me. We don't want users to deploy additional Prometheus objects in the openshift-monitoring or openshift-user-workload-monitoring namespaces either.

Comment (bburt-rh) [reply modules/monitoring-support-considerations.adoc:21]: @juzhao Do you agree with @simonpasquier? If so, please give this PR an lgtm. If not, why not?

File: modules/monitoring-exposing-custom-application-metrics-for-horizontal-pod-autoscaling.adoc (hunk 3 above)
Comment (gabriel-rh) [reply modules/monitoring-exposing-custom-application-metrics-for-horizontal-pod-autoscaling.adoc]: I don't see this change in the file?

**Review by gabriel-rh — APPROVED** [review]
/lgtm

File: modules/monitoring-support-considerations.adoc (hunk 4 above)
Comment (juzhao) [reply modules/monitoring-support-considerations.adoc:21]: thanks, LGTM

**Pull request merged**